src/module-elasticsuite-core/Index/Mapping/Field.php

```diff
@@ -354,19 +353,26 @@ private function getFieldAnalyzers()
      *
      * @return array
      */
-    private function getPropertyConfig($analyzer = self::ANALYZER_UNTOUCHED)
+    private function getPropertyConfig($analyzer = self::ANALYZER_UNTOUCHED): array
     {
         $fieldMapping = ['type' => $this->getType()];
 
-        if ($this->getType() == self::FIELD_TYPE_TEXT && $analyzer == self::ANALYZER_UNTOUCHED) {
-            $fieldMapping['type']  = self::FIELD_TYPE_KEYWORD;
-        } elseif ($this->getType() == self::FIELD_TYPE_TEXT) {
-            $fieldMapping['analyzer'] = $analyzer;
-            if ($analyzer === self::ANALYZER_SORTABLE) {
-                $fieldMapping['fielddata'] = true;
-            }
-        } elseif ($this->getType() == self::FIELD_TYPE_DATE) {
-            $fieldMapping['format'] = implode('||', $this->dateFormats);
+        switch ($this->getType()) {
```

Comment:
I'm not fan of switch/case, I prefer the original syntax with if/elseif (and it would also allow us to have a relevant git history).

Reply:
For me better use "switch/case" instance of many if/else. You're right about git history, but, in my opinion, we are losing in the code style and structure.